presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,245 @@
+    {
+        List<Object> includeValues = new ArrayList<>();
+        List<Object> exlucdeValues = new ArrayList<>();
+        for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
+            checkState(!range.isAll());
+            if (range.isSingleValue()) {
+                includeValues.add(range.getLow().getValue());
+            }
+
+            if (!range.getLow().isLowerUnbounded() && range.getLow().getBound() == ABOVE) {
+                Object value = range.getLow().getValue();
+                if (!exlucdeValues.contains(value)) {
+                    exlucdeValues.add(value);
+                }
+            }
+
+            if (!range.getHigh().isUpperUnbounded() && range.getHigh().getBound() == BELOW) {
+                Object value = range.getHigh().getValue();
+                if (!exlucdeValues.contains(value)) {
+                    exlucdeValues.add(value);
+                }
+            }
+        }
+
+        for (Object value : exlucdeValues) {
+            builder.mustNot(new MatchQueryBuilder(columnName, getValue(type, value)));
+        }
+
+        if (includeValues.size() == 1) {
+            builder.must(new MatchQueryBuilder(columnName, getValue(type, getOnlyElement(includeValues))));
+        }
+        else {
+            for (Object value : includeValues) {
+                builder.should(new MatchQueryBuilder(columnName, getValue(type, value)));
+            }
+        }
+        return builder;
+    }
+
+    private QueryBuilder buildTermQuery(BoolQueryBuilder builder, String columnName, Domain domain, Type type)
+    {
+        List<Object> values = new ArrayList<>();
+        for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
+            checkState(!range.isAll());
+            if (range.isSingleValue()) {
+                values.add(range.getLow().getValue());
+            }
+            else {
+                if (!range.getLow().isLowerUnbounded()) {
+                    switch (range.getLow().getBound()) {
+                        case ABOVE:
+                            builder.must(new RangeQueryBuilder(columnName).gt(getValue(type, range.getLow().getValue())));
+                            break;
+                        case EXACTLY:
+                            builder.must(new RangeQueryBuilder(columnName).gte(getValue(type, range.getLow().getValue())));
+                            break;
+                        case BELOW:
+                            throw new IllegalArgumentException("Low marker should never use BELOW bound");
+                        default:
+                            throw new AssertionError("Unhandled bound: " + range.getLow().getBound());
+                    }
+                }
+                if (!range.getHigh().isUpperUnbounded()) {
+                    switch (range.getHigh().getBound()) {
+                        case ABOVE:
```

Comment:
move the `ABOVE` case right above the `default` case, so the failure cases are together.